python_ta/__init__.py

```diff
@@ -109,7 +109,9 @@ def _check(
 
     # Try to check file, issue error message for invalid files.
     try:
-        for locations in _get_valid_files_to_check(module_name):
+        files_to_check = list(_get_valid_files_to_check(module_name))
```

Comment:
This isn't exactly the right place to do this because `file_to_check` might be a sequence of *directories*, not just files, and I'd like us to handle the case when the folder exists but it contains no files (also not showing the HTML report in this case). That's why there's an inner loop `for file_pay in get_file_paths(locations)` below.

I'd suggest using a boolean flag like "file found" or something and using that to indicate whether at least one file was actually checked or not.